docs/proposals/14-adminkubeconfig-subresource.md

```diff
@@ -0,0 +1,103 @@
+---
+title: Dynamic kubeconfig generation for Shoot clusters
+gep-number: 14
+creation-date: 2021-04-02
+status: implementable
+authors:
+- "@mvladev"
+reviewers:
+- "@rfranzke"
+---
+
+# GEP-14: Dynamic kubeconfig generation for Shoot clusters
+
+## Table of Contents
+
+- [GEP-14: Dynamic kubeconfig generation for Shoot clusters](#gep-14-dynamic-kubeconfig-generation-for-shoot-clusters)
+  - [Table of Contents](#table-of-contents)
+  - [Summary](#summary)
+  - [Motivation](#motivation)
+    - [Goals](#goals)
+    - [Non-Goals](#non-goals)
+  - [Proposal](#proposal)
+  - [Alternatives](#alternatives)
+
+## Summary
+
+This `GEP` introduces new  `Shoot` subresource called `AdminKubeConfigRequest` allowing for users to dynamically generate a short-lived `kubeconfigs` that can be used to access the `Shoot` cluster as `cluster-admin`.
+
+## Motivation
+
+Today, when access to the created `Shoot` clusters is needed, a `kubeconfig` with static token credentials is used. This static token is in the `system:masters` group, granting it `cluster-admin` privileges. The `kubeconfig` is generated when the cluster is reconciled, stored in `ShootState` and replicated in the `Project`'s namespace in a `Secret`. End-users can fetch the secret and use the `kubeconfig` inside it.
+
+There are several problems with this approach:
+
+- The token in the `kubeconfig` does not have any expiration, so end-users have to issue request a `kubeconfig` credential rotation call manually.
```

Comment:
s/end-users have to issue request a `kubeconfig` credential rotation call manually/end-users have to request a `kubeconfig` credential rotation if they want revoke the token